PurchasesCoreSwift/SubscriberAttributes/AttributionDataMigrator.swift

```diff
@@ -61,90 +61,91 @@ private extension AttributionDataMigrator {
         return networkSpecificSubscriberAttributes
     }
 
-    func convertMParticleAttribution(_ data: [String: Any]) -> [String: Any] {
-        var convertedAttribution: [String: Any] = [:]
-        if let value = data[AttributionKey.MParticle.id.rawValue] {
+    func convertMParticleAttribution(_ data: [String: Any]) -> [String: String] {
+        var convertedAttribution: [String: String] = [:]
+        if let value = data[AttributionKey.MParticle.id.rawValue] as? String {
             convertedAttribution[SpecialSubscriberAttributes.mpParticleID] = value
         }
-        if let value = data[AttributionKey.networkID.rawValue] {
+        if let value = data[AttributionKey.networkID.rawValue] as? String {
             convertedAttribution[SpecialSubscriberAttributes.mpParticleID] = value
         }
         return convertedAttribution
     }
 
-    func convertBranchAttribution(_ data: [String: Any]) -> [String: Any] {
-        var convertedAttribution: [String: Any] = [:]
-        if let value = data[AttributionKey.Branch.channel.rawValue] {
+    func convertBranchAttribution(_ data: [String: Any]) -> [String: String] {
+        var convertedAttribution: [String: String] = [:]
+        if let value = data[AttributionKey.Branch.channel.rawValue] as? String {
             convertedAttribution[SpecialSubscriberAttributes.mediaSource] = value
         }
-        if let value = data[AttributionKey.Branch.campaign.rawValue] {
+        if let value = data[AttributionKey.Branch.campaign.rawValue] as? String {
             convertedAttribution[SpecialSubscriberAttributes.campaign] = value
         }
         return convertedAttribution
     }
 
     // swiftlint:disable cyclomatic_complexity
-    func convertAppsFlyerAttribution(_ data: [String: Any]) -> [String: Any] {
+    func convertAppsFlyerAttribution(_ data: [String: Any]) -> [String: String] {
         var fixedData = data
-        if let innerDataObject = fixedData[AttributionKey.AppsFlyer.dataKey.rawValue] as? [String: Any?] {
+        // TODO confirm casting all to strings makes sense
+        if let innerDataObject = fixedData[AttributionKey.AppsFlyer.dataKey.rawValue] as? [String: String?] {
             if fixedData[AttributionKey.AppsFlyer.statusKey.rawValue] != nil {
                 for (key, value) in innerDataObject {
                     fixedData[key] = value
                 }
             }
         }
 
-        var convertedAttribution: [String: Any] = [:]
+        var convertedAttribution: [String: String] = [:]
 
-        if let value = fixedData[AttributionKey.networkID.rawValue] {
+        if let value = fixedData[AttributionKey.networkID.rawValue] as? String {
```

Comment:
it appears that all of the values we persist are strings, but can someone confirm?

Reply:
@vegaro might know more, since he implemented this fairly recently

Reply:
i'll make a note to follow up with him

Reply:
@vegaro when you get a chance ^^

Reply:
Sorry, I just saw this. I think as long as we keep attributionData as [String: Any] (which you are doing already) we should be fine. All the converted subscriber attributes are strings, so this is actually an improvement 😄